Framework/Kernel/inc/MantidKernel/MagneticIon.h

```diff
@@ -20,9 +20,8 @@ struct MANTID_KERNEL_DLL MagneticIon {
   /// Default constructor
   MagneticIon();
   /// Construct the Ion with data
-  MagneticIon(const std::string symbol, const uint16_t charge,
-              const double j0[8], const double j2[8], const double j4[8],
-              const double j6[8]);
+  MagneticIon(const char *symbol, const uint16_t charge, const double j0[8],
+              const double j2[8], const double j4[8], const double j6[8]);
```

Comment:
I think that can be done, but it's a bit more work than replacing `std::string` with `char[]` or `const char *`. The constructor copies arrays j0, j2, j4 and j6 into std::vectors, which is also the container that anything using MagneticIon expects.

We'e also had some discussion whether it would be better to move this data out of the code and into a separate file.